lib/fog/aliyun/models/compute/security_group.rb

```diff
@@ -4,82 +4,84 @@ module Fog
   module Compute
     class Aliyun
       class SecurityGroup < Fog::Model
-        # {"SecurityGroup"=>[{"CreationTime"=>"2015-08-10T03:21:02Z", 
-        #   "SecurityGroupId"=>"sg-25mexinlu", "Description"=>"", "SecurityGroupName"=>"bosh", 
-        #   "VpcId"=>"vpc-25mj6mguq"}}
-
-        identity  :id,                    :aliases =>"SecurityGroupId"
-
-        attribute :name,                  :aliases => "SecurityGroupName"
-        attribute :description,           :aliases => "Description"
-        attribute :created_at,            :aliases => "CreationTime"
-        attribute :vpc_id,                :aliases => "VpcId"
+        identity  :id,           :aliases =>"SecurityGroupId"
+        attribute :name,         :aliases => "SecurityGroupName"
+        attribute :description,  :aliases => "Description"
+        attribute :created_at,   :aliases => "CreationTime"
+        attribute :vpc_id,       :aliases => "VpcId"
 
         def vpc
-            requires :vpc_id
-            $vpc=Fog::Compute::Aliyun::Vpcs.new(:service=>service).all('vpcId'=>vpc_id)[0]
+          requires :vpc_id
+          service.vpcs.get(vpc_id)
+          end
         end
 
         def security_group_rules
-            requires :id
-            Fog::Compute::Aliyun::SecurityGroupRules.new(:service=>service).get(id)
+          requires :id
+          Fog::Compute::Aliyun::SecurityGroupRules.new(:service=>service).get(id)
         end
 
         def save(options={})
-            options[:vpcId] = vpc_id if vpc_id
-            options[:name] = name if name
-            options[:description] = description if description
-            data = Fog::JSON.decode(service.create_security_group(options).body)
-            true
+          options[:vpcId] = vpc_id if vpc_id
+          options[:name] = name if name
+          options[:description] = description if description
+          data = Fog::JSON.decode(service.create_security_group(options).body)
```

Comment:
This encoding and decoding should be up in the request itself. The model should work on top of a ruby hash and deal with business logic only.